models/taskRequests.js

```diff
@@ -435,6 +439,27 @@ const approveTaskRequest = async (taskRequestId, user) => {
   }
 };
 
+const rejectTaskRequest = async (taskRequestId, authenticatedUserId) => {
+  const taskRequestDoc = taskRequestsCollection.doc(taskRequestId);
+  const taskRequestData = (await taskRequestDoc.get()).data();
+  if (!taskRequestData) {
+    return { taskRequestNotFound: true };
+  }
+
+  if (
+    taskRequestData.status === TASK_REQUEST_STATUS.APPROVED ||
+    taskRequestData.status === TASK_REQUEST_STATUS.DENIED
+  ) {
+    return { isTaskRequestInvalid: true };
+  }
+  const updatedTaskRequest = {
+    status: TASK_REQUEST_STATUS.DENIED,
+    lastModifiedBy: authenticatedUserId,
+    lastModifiedAt: Date.now(),
+  };
+  await taskRequestDoc.update(updatedTaskRequest);
+  return { taskRequest: { ...taskRequestData, ...updatedTaskRequest } };
+};
```

Comment:
- please add a jsdoc comment mentioning the return type of this function.
- we are returning very different values for different branches of the function, should we have a consistent return type instead?

Reply:
Added js doc

Reply:
- should we have a consistent return value (the value being returned can differ for different branches)?
```js
{
   taskRequestNotFound: true / false,
   isTaskRequestInvalid: true / false,
   taskRequest: null / <value returned in happy flow>
}
```
- This will make it easy to consume in a typescript application where union types can get tricky to use.

Reply:
Updated the doc.